LambdaLinq/Controllers/HomeController.cs

```diff
@@ -31,17 +36,29 @@ public ActionResult Search()
         }
 
         [HttpPost]
-        public ActionResult Search(Book searchBook)
+        public ActionResult Search(SearchViewModel searchBook)
         {
-            //TODO: add logic to return all books if the search is empty
 
-            var foundBooks = db.Books.Where(b => b.Author == searchBook.Author);
+            var foundBooks = db.Books.Where(b => b != null);
+
+            //TODO: add logic to return all books if the search is empty
+            if (searchBook.Author != null)
+            {
+                foundBooks = db.Books.Where(b => b.Author == searchBook.Author);
+            }
 
             //TODO: add logic to search by Title (Note: you will need to adjust the View and ViewModel)
+            else if (searchBook.Title != null)
+            {
+                foundBooks = db.Books.Where(b => b.Title.ToLower() == searchBook.Title.ToLower());
+            }
 
             //TODO: add logic to return a search on price between a low and high number (Note: you will need to adjust the View and ViewModel)
-
-            return View("SearchResult", foundBooks);
+            else if (searchBook.PriceRangeMin > 0 && searchBook.PriceRangeMax > 0)
```

Comment:
a better condition would be
if (searchBook.PriceRangeMin > 0 && searchBook.PriceRangeMax > searchBook.PriceRangeMin)